pkg/sql/colexec/preinsertsecondaryindex/preinsertsecondaryindex.go

```diff
@@ -110,7 +111,8 @@ func (preInsertSecIdx *PreInsertSecIdx) Call(proc *process.Process) (vm.CallResu
 	preInsertSecIdx.ctr.buf.SetVector(indexColPos, vec)
 	preInsertSecIdx.ctr.buf.SetRowCount(vec.Length())
 
-	vec, err = util.CompactPrimaryCol(inputBat.Vecs[pkPos], bitMap, proc)
+	vec = proc.GetVector(*inputBat.Vecs[pkPos].GetType())
```

Comment:
This too.    If you really need a vector, not a batch, put the vector in your ctr.   Do not lose it.   Make a comment in types.go Use it, reuse it, and free it at the right time.